src/components/common/UnclaimedTransfers/UnclaimedTransfers.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import { useColonyContext } from '~hooks';
+import { ADDRESS_ZERO } from '~constants';
+
+import UnclaimedTransfersItem from './UnclaimedTransfersItem';
+
+import styles from './UnclaimedTransfers.css';
+import { ColonyFundsClaim } from '~gql';
+
+const displayName = 'common.UnclaimedTransfers';
+
+const MSG = defineMessages({
+  title: {
+    id: `${displayName}.title`,
+    defaultMessage: 'Incoming funds for {colony}',
+  },
+  loadingData: {
+    id: `${displayName}.loadingData`,
+    defaultMessage: 'Loading token transfers...',
+  },
+});
+
+const UnclaimedTransfers = () => {
+  const { colony } = useColonyContext();
+  const { profile, fundsClaims, chainFundsClaim, tokens } = colony || {};
+  const { items: claims = [] } = fundsClaims || {};
+
+  /*
+   * @NOTE We have to do some very heavy lifting (more or less) here due to us
+   * not being able to use Apollo's cache, so we want to short-circuit early
+   * in order to not waste any computing resources unecesarily
+   */
```

Comment:
This logic seem to be duplicated here and in `ColonyUnclaimedTransfers`. Could be extracted into a hook potentially?

Reply:
Make sense. Will do that.